distributed/shuffle/_worker_plugin.py

```diff
@@ -31,146 +32,59 @@
 logger = logging.getLogger(__name__)
 
 
-class ShuffleWorkerPlugin(WorkerPlugin):
-    """Interface between a Worker and a Shuffle.
-
-    This extension is responsible for
-
-    - Lifecycle of Shuffle instances
-    - ensuring connectivity between remote shuffle instances
-    - ensuring connectivity and integration with the scheduler
-    - routing concurrent calls to the appropriate `Shuffle` based on its `ShuffleID`
-    - collecting instrumentation of ongoing shuffles and route to scheduler/worker
-    """
-
-    worker: Worker
-    shuffles: dict[ShuffleId, ShuffleRun]
+class ShuffleRunManager(Mapping[ShuffleId, ShuffleRun]):
```

Comment:
Is it truly helpful to define this as a mapping?

Reply:
It makes testing a tad easier (i.e., existing tests assume a mapping) but apart from that it doesn't matter. I'm open to adjusting tests if we want to get rid of the `Mapping`.

Reply:
as long as we don't rely on it too strongly, I'm OK with it

Reply:
I like `Mapping` for simple stuff that doesn't think too much. This thing is a living thing that makes decisions.

Reply:
I've thought about it and will remove the `Mapping` to prevent anybody from using it. What is considered to be inside the mapping (i.e., _active_ shuffle runs) is not super obvious.